code/glTFExporter.cpp

```diff
@@ -753,9 +810,12 @@ inline void ExtractAnimationData(Asset& mAsset, std::string& animId, Ref<Animati
     //-------------------------------------------------------
     // Extract rotation parameter data
     if(nodeChannel->mNumRotationKeys > 0) {
-        C_STRUCT aiQuaternion* rotationData = new aiQuaternion[nodeChannel->mNumRotationKeys];
+        vec4* rotationData = new vec4[nodeChannel->mNumRotationKeys];
```

Comment:
`nodeChannel->mRotationKeys` is represented as a quaternion internally to assimp. That is `C_STRUCT aiQuaternion`. However when it is exported we need it as a simple vector of length 4.

`rotationData` is only used to export the `mRotationKeys` to the buffer. During this export into the buffer, it issues the [] operator which does not exist for the `aiQuaternion` struct.

This also makes sure that it is output to the buffer in the order of `x, y, z, w` which is the glTF specification.